### PR TITLE
feat(flow): persist complete CTS inventories

### PR DESCRIPTION
**Pull request by ThePlenkov**
## Summary

- retain every CTS object in a deterministic manifest inventory before source-format filtering
- write one .adt/tr/<TR>.json descriptor for every discovered request and task
- resolve supported LIMU leaves such as METH, CLSD, CPUB, and FUNC through their CLAS/FUGR repository owners while preserving original CTS identity
- expose URI-free inventory metadata through CLI, MCP, and REST boundaries

## Why

Unsupported transport entries currently disappear from source checkout state. A task that has no materializable source therefore cannot be identified and retried when support expands. Persisting the complete task-level inventory makes that state durable and allows later materialization without losing provenance.

## Validation

- ADK full suite passed
- adt-flow full suite passed
- adt-cli: 306 passed, 3 todo
- adt-mcp: 91 passed
- adt-server: 70 passed
- lint passed for adk, adt-flow, adt-cli, adt-mcp, and adt-server
- affected builds passed
- OpenSpec strict validation passed

## Notes

The adt-cli typecheck target still reports pre-existing errors in adt-plugin-abapgit and unrelated object-builder command types; adk and adt-flow typechecks pass, and all affected builds succeed.

---
## Summary by cubic
Persist the complete CTS object inventory and write one deterministic `.adt/tr/<TR>.json` per discovered request/task; exact-head fast path now validates all scoped transport descriptors to avoid stale reuse.

- Resolves supported LIMU leaves (`METH`, `CLSD`, `CPUB`, `FUNC`) through their `CLAS`/`FUGR` repository owners while preserving the original CTS identity; entries may include `repositoryObject`, and grouping/deduplication use the repository owner.
- Adds `inventory` to the `adk` manifest and `adt-flow` transport descriptor; each descriptor keeps only its objects (`sourceTransport === <TR>`). CLI/`adt-mcp`/`adt-server` expose this inventory with SAP ADT URIs stripped. `@adt-flow` exports `transportObjectInventoryEntrySchema`; `transportDescriptorSchema.inventory` is optional in schema but required by the fast path.
- Exact-head fast path requires descriptors for every scoped transport with matching `requestedTransports`, `scopeTransports`, digests, and a present `inventory`; it also verifies each inventory entry’s `sourceTransport` matches its descriptor. Missing/mismatched descriptors fall back to a rebuild.
- `adt-flow` updates all scoped descriptors during task-only checkouts and reuses existing matching descriptors without overwrite.

- Migration:
  - Consumers of `adk`/`adt-flow` manifest data must handle `repositoryObject` and `inventory`.
  - Tools must handle multiple `.adt/tr/<TR>.json` descriptors per scope.
  - Do not depend on SAP ADT URIs in CLI/MCP/REST responses; they are removed.
  - First head checkout after upgrade may regenerate transport descriptors; exact-head fast path requires inventoryed scoped descriptors.

<sup>Written for commit 50eb8fe6bd7924d6ebccdb1cd09a3d4132b3410f. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/abapify/adt-cli/pull/182?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/adt-flow/src/service.ts [hunk 1]
```diff
@@ -298,6 +302,7 @@ async function exactHeadFastPath(
         !descriptor ||
         descriptor.configDigest !== configDigest ||
         descriptor.formatDigest !== formatDigest ||
+        descriptor.inventory === undefined ||
```
[the remark this answers is not in the document]
Comment (ThePlenkov) [reply packages/adt-flow/src/service.ts:305]: The invalidation is required for this schema addition. A legacy descriptor has no complete CTS inventory, and the exact-head path cannot reconstruct unsupported or selector-excluded objects without querying CTS. Reusing it would falsely claim the new durable inventory contract is satisfied. The next online checkout performs the explicit migration by rebuilding and writing inventory-bearing descriptors.

File: packages/adk/src/transport-source-manifest.ts [hunk 2]
```diff
@@ -628,36 +640,177 @@ export function selectTransportSourceVersions( // NOSONAR - SAP feed-order selec
...
+      ? repositoryNameFromUri(reference.uri, ownerType)
+      : undefined;
+    if (ownerType && ownerName) {
+      return {
+        pgmid: 'R3TR',
+        type: ownerType,
+        name: ownerName,
+        // A deleted leaf changes its owner. Only REPS preserves the legacy
+        // whole-program deletion behavior when no R3TR parent is present.
+        isDeleted: rawType === 'REPS' && reference.isDeleted,
```
[the remark this answers is not in the document]
Comment (ThePlenkov) [reply packages/adk/src/transport-source-manifest.ts:688]: Verified as intentional owner semantics. A deleted LIMU method/section/function leaf means its repository owner changed; the class or function group itself still exists and must be compared through normal source history. Treating the owner as deleted would remove the whole class/group for a leaf deletion. REPS is the exception because it represents the whole program source. No code change.

File: packages/adt-flow/tests/service.test.ts [hunk 3]
```diff
@@ -295,6 +306,53 @@ describe('transport checkout', () => {
...
+    current.inventory.push({
+      pgmid: 'LIMU',
+      type: 'ZZZZ',
+      name: 'ZUNSUPPORTED',
+      wbtype: 'ZZZZ',
+      uri: '/sap/bc/adt/repository/informationsystem/objectproperties/values',
+      objFunc: '',
+      sourceTransport: 'DEVK900002',
+    });
+    current.entries.push(unsupportedEntry('ZUNSUPPORTED'));
```
[the remark this answers is not in the document]
Comment (ThePlenkov) [reply packages/adt-flow/tests/service.test.ts]: Fixed in 50eb8fe6. The helper now accepts sourceTransport and the unsupported entry uses DEVK900002, matching its inventory record.

File: openspec/changes/add-adt-flow-transport-checkout/tasks.md [hunk 4]
```diff
@@ -59,3 +59,10 @@
 - [x] 8.1 Add a failing indexed-flow test proving an `added` parent-boundary manifest preserves the prior task state in base mode.
 - [x] 8.2 Reuse the verified present descriptor and owned files as the incremental base without source or mutable-state reads.
 - [x] 8.3 Prove an unindexed format-recognizable tree remains unchanged in base mode and is adopted by head checkout.
+
+## 9. Preserve complete CTS inventory for later materialization
+
+- [x] 9.1 Add failing tests proving selector-excluded and unsupported CTS objects remain in the manifest inventory.
+- [x] 9.2 Persist one `.adt/tr/<TR>.json` descriptor per discovered request/task with concrete object provenance.
+- [x] 9.3 Resolve supported LIMU leaves through their `wbtype` and ADT owner URI while preserving the original CTS identity.
+- [ ] 9.4 Prove an inventory-only task survives the consumer's state-only publication path and can be retried after support expands.
```
[the remark this answers is not in the document]
Comment (ThePlenkov) [reply openspec/changes/add-adt-flow-transport-checkout/tasks.md]: Completed in 50eb8fe6. The downstream consumer regression proves inventory-only state is published without creating a review record and remains retryable; task 9.4 now records that cross-repository proof without embedding private consumer identifiers in the public change.
